src/notification.cpp

```diff
@@ -299,8 +458,10 @@ void Notification::close()
 */
 void Notification::checkActionInvoked(uint id, QString actionKey)
 {
-    if (id == replacesId_ && actionKey == "default") {
-        emit clicked();
+    if (id == replacesId_) {
+        if (actionKey == DEFAULT_ACTION_NAME) {
```

Comment:
They're actually logically separate - the first condition is applicability to this notification, the second is switching on which action was invoked.  If there is further modification to this code, it will be inside the first condition.